### PR TITLE
Datalab testkit 2.0

### PR DESCRIPTION
**Pull request by awisema**
- [X] Testkit as separate driver.
- [X] Testing for the SPM

### CODE REVIEW

File: drivers/datalab_testkit/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+of HMIS inputs and the outputs from running the HUD APR, CAPER, CE-APR, and SPMs.
+
+As of version 2.0, there is not a standardized representation for the SPM results,
+and the test kit contains a `.xlsx` containing the test results.
+`DatalabTestkit::TestkitSpmXlsxToCsv` is a translator to generate CSVs in the
+format used for Warehouse HUD report tests from the Excel file:
+
+<pre>
+DatalabTestkit::TestkitSpmXlsxToCsv.new(<i>directory</i>).convert(<i>excel_filename.xlsx</i>)
+</pre>
```
Comment (eanders) [inline drivers/datalab_testkit/README.md:13]: Maybe this should be in markdown instead of html?

**Review by eanders — COMMENTED** [review]
This looks good, but I'm a bit concerned the tests didn't finish after 5+ hours.  I'm thinking we may need to streamline the data loading.  Maybe turn the imports into a single SQL file we can load?  That's not as nice since this runs through the importer, but it might significantly increase test speed.

File: drivers/datalab_testkit/README.md (hunk 1 above)
Comment (awisema) [reply drivers/datalab_testkit/README.md:13]: Maybe -- I couldn't get markup (the italics) to work inside a code region in markdown. At least in my markdown tool and GH, the html works

**Comment by awisema** [comment]
Yeah, that's not going to scale to the rest of the HUD reporting if it take 5hrs for one test... Unfortunate that the fixpoint runs out of RAM... I'll play with some ideas and see what we can do -- maybe I can combine all the CSVs, either by merging and de-duping (they are all distinct projects, so only if things like organizations overlap will that even be needed), or do an import and then an export?

**Comment by eanders** [comment]
I was wondering about merging the CSVs too.  I think you are correct, only Export.csv and Organization.csv should have any duplicates.  The rest should just be able to be concatenated.  I don't think we want to do an import/export since that'll change a bunch of identifiers.

**Pull request merged**